### ISSUE TITLE
record counts incorrectly report some records as active update instead of active new

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/xcmetadataservicestoolkit#457]
JIRA issue created by: jbrand
Originally opened: 2012-02-20 03:22 PM

Issue body:
 (nt)

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-02-20 03:22 PM

Comment body:

<p>
	During the running of the T0 test, I noticed 21 upd_act_cnt records reported.&nbsp; This is wrong.&nbsp; All records were seen for the 1st time during t0.&nbsp; The reason appears to be that these holdings records were marked HELD because they were waiting on a bib, then when the bib arrived the status changed to active.&nbsp; The code in RecordCounts.java looks to see if the previous status was NULL, if so it KNOWS it is an active new record.&nbsp; If it had a previous status of H it thinks it saw it already.</p>

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-02-20 05:56 PM

Comment body:

Most likely this is next for Chris, to bridge the gap between the 1.3 work and release and when Jennifer is able to finish defining 001/003 and embedded holdings work.

John to write up what was decided at meeting today.

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-02-20 06:44 PM

Comment body:

<p>
	We agreed to look into separating the &#39;held&#39; status from the deleted/active/new status.&nbsp;</p>
<p>
	An open question is whether a different path is taken through transformation based on: active/updated/new.</p>
<p>
	&nbsp;</p>

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-02-22 09:12 PM

Comment body:

<p>
	<br />
	The answer to the open question: &quot;<span id="BugEvents">whether a different path is taken through transformation based on: active/updated/new</span>&quot;</p>
<p>
	The answer:&nbsp; No, whether a record has been seen or not by the service plays no part in how the records are transformed.</p>
<p>
	&nbsp;</p>

**Comment by patrickzurek** [comment]
JIRA Coment by user: dlindahl
JIRA Timestamp: 2012-02-24 06:51 PM

Comment body:

I am not sure that this is correct.  If there are previously created successors, they will need to be updated.  If there are none, then they need to be created.

I know we don't support embedded holdings yet (or do we)? But if/when we do, this needs to be true:
If we have some previously created successor holding records that were produced as a result of input bib records with embedded holdings, we need to be able to delete previously created successor holdings.

e.g. In the case where a bib record had 7 embedded holdings and then an updated version of the bib now has only 6 embedded holdings, the service will need to delete an output holding record.

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-02-27 02:31 PM

Comment body:

Maybe a separate issue should be broken out for marctoxctransformation service work that is needed to do things like:


e.g. In the case where a bib record had 7 embedded holdings and then an updated version of the bib now has only 6 embedded holdings, the service will need to delete an output holding record.

Or maybe this is already part of what is being written and spec'd out for embedded holdings?

In short, 'where do we go from here?'

I propose that in any event, this fb issue is only used to track work needed to address the subject line above.

**Comment by patrickzurek** [comment]
JIRA Coment by user: dlindahl
JIRA Timestamp: 2012-02-28 07:56 PM

Comment body:

I agree that embedded holdings is a separate issue, and we will leave it alone for now.  This issue remains as just being about "active update" vs "active new"

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-05-10 11:57 AM

Comment body:

At this point, this becomes a documentation (not coding change) task that clearly details what is being counted by the MST, by Service and "buckets", and at various points in the system processing, taking the "time" of processing and state of record into account. After we know what it is doing, we can then work on the reconcile process with Drupal, etc.

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-05-10 02:13 PM

Comment body:

Existing wiki record counts documents:

http://code.google.com/p/xcmetadataservicestoolkit/wiki/LoggingRecordCounts
http://code.google.com/p/xcmetadataservicestoolkit/wiki/verifyRecordCounts

Probably need to expand one of these.

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-05-31 09:53 AM

Comment body:

John, I assume from your last comment that this is not done?  And that you would like this moved to 1.5

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-05-31 09:55 AM

Comment body:

The only thing we are doing with this issue is adding to documentation.  1.5 sounds like an agreeable time frame for me to work on this.

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-06-27 03:30 PM

Comment body:

I started adding to the documentation here:

http://code.google.com/p/xcmetadataservicestoolkit/wiki/verifyRecordCounts

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbrand
JIRA Timestamp: 2012-06-28 03:42 PM

Comment body:

I documented that which I know here: http://code.google.com/p/xcmetadataservicestoolkit/wiki/verifyRecordCounts

**Comment by patrickzurek** [comment]
Issue resolved: 2012-06-28 03:42 PM
